search/search_repository_blackbox_test.go

```diff
@@ -213,48 +203,65 @@ func (s *searchRepositoryBlackboxTest) TestSearchFullText() {
 		})
 	})
 
-	s.T().Run("Filter with parent-exists filter", func(t *testing.T) {
+	s.T().Run("with parent-exists filter", func(t *testing.T) {
 
 		t.Run("no link created", func(t *testing.T) {
 			// given
-			fxt := s.getTestFixture()
+			fxt := tf.NewTestFixture(t, s.DB, tf.WorkItems(3))
 			// when
 			filter := fmt.Sprintf(`{"$AND": [{"space": "%s"}]}`, fxt.Spaces[0].ID)
 			parentExists := false
 			res, count, err := s.searchRepo.Filter(context.Background(), filter, &parentExists, nil, nil)
 			// then both work items should be returned
 			require.Nil(t, err)
-			assert.Equal(t, uint64(2), count)
-			assert.Equal(t, 2, len(res))
+			assert.Equal(t, uint64(3), count)
+			assert.Equal(t, 3, len(res))
 		})
 
 		t.Run("link created", func(t *testing.T) {
 			// given
-			fxt := s.getTestFixture(tf.WorkItemLinks(1, func(fxt *tf.TestFixture, idx int) error {
-				fxt.WorkItemLinks[0].LinkTypeID = fxt.WorkItemLinkTypes[0].ID
-				fxt.WorkItemLinks[0].SourceID = fxt.WorkItems[0].ID
-				fxt.WorkItemLinks[0].TargetID = fxt.WorkItems[1].ID
-				return nil
-			}))
+			fxt := tf.NewTestFixture(t, s.DB,
+				tf.WorkItemLinkTypes(1, func(fxt *tf.TestFixture, idx int) error {
+					// need an explicit 'parent-of' type of link
+					fxt.WorkItemLinkTypes[idx].ForwardName = link.TypeParentOf
+					fxt.WorkItemLinkTypes[idx].Topology = link.TopologyTree
+					return nil
+				}),
+				tf.WorkItems(3),
+				tf.WorkItemLinks(1, func(fxt *tf.TestFixture, idx int) error {
+					fxt.WorkItemLinks[0].LinkTypeID = fxt.WorkItemLinkTypes[0].ID
+					fxt.WorkItemLinks[0].SourceID = fxt.WorkItems[0].ID
+					fxt.WorkItemLinks[0].TargetID = fxt.WorkItems[1].ID
```

Comment:
Those three lines exactly mimic what is already done for you in the `makeWorkItemLinks` function:

1. Always take the first link type (`fxt.WorkItemLinkTypes[0].ID`)
2. Link from `fxt.WorkItems[2*idx].ID` to `fxt.WorkItems[2*idx+1].ID`